public/javascripts/controllers/buyer-ctrl.js

```diff
@@ -168,10 +168,13 @@ angular.module('jasmic.controllers')
             /**
              * Fetches the units that user can select
              */
-             UnitsFactory.query({un_unit_class: "Weight"}, function(units){
-                    $scope.units = units;
+
+            UnitsFactory.query({un_unit_class: "Weight"}, function(units){
+                $scope.units = units;
             });
 
+            $scope.units = UnitsFactory.query({});
```

Comment:
@nick22891 this line will overwrite the `units` already fetched from the above function call.